api/src/main/java/org/apache/iceberg/io/FileAppender.java

```diff
@@ -45,4 +46,11 @@ default void addAll(Iterable<D> values) {
    * @return the length of this file. Only valid after the file is closed.
    */
   long length();
+
+  /**
+   * @return list of offsets for blocks of a file. Only valid after the file is closed.
+   */
+  default List<Long> offsetRanges() {
```

Comment:
What about `splitOffsets`? I think that is more clear. This should also have clear Javadoc about what is returned and how it may be used for producing splits.